src/prefect/tasks/spacy/spacy_tasks.py

```diff
@@ -0,0 +1,204 @@
+
+import spacy
+
+
+class SpacyNLP(Task):
+    """
+    Task for processing text with a spaCy pipeline.
+
+    Args:
+        - text (unicode, optional): string to be processed, can be provided during construction
+            or when task is run
+        - nlp (spaCy text processing pipeline, optional): a custom spaCy text
+            processing pipeline, if provided, this pipeline will be used instead
+            of being created from spacy_model_name
+        - spacy_model_name (str, optional): name of the spaCy language model, default
+            model is 'en_core_web_sm', will be ignored if nlp is provided
+        - disable (List[str], optional): list of pipeline components
+            to disable, only applicable to pipelines loaded from spacy_model_name
+        - component_cfg (dict, optional): a dictionary with extra keyword
+            arguments for specific components, only applicable to pipelines loaded from
+            spacy_model_name
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        text: str = u"",
+        nlp=None,
+        spacy_model_name: str = "en_core_web_sm",
+        disable: list = [],
+        component_cfg: dict = {},
+        **kwargs
+    ):
+        self.text = text
+
+        ## load spacy model
+        if nlp:
+            self.nlp = nlp
+        else:
+            try:
+                self.nlp = spacy.load(
+                    spacy_model_name, disable=disable, component_cfg=component_cfg
+                )
+            except IOError:
+                raise ValueError("spaCy model %s not found." % spacy_model_name)
+
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("text")
+    def run(self, text: str = u""):
+        """
+        Task run method. Creates a spaCy document.
+
+        Args:
+            - text (unicode, optional): text to be processed
+
+        Returns:
+            - Doc: spaCy document
+        """
+        doc = self.nlp(text)
+
+        return doc
+
+
+class SpacyTagger(Task):
+    """
+    Task for returning tagger from a spaCy pipeline.
+
+    Args:
+        - nlp (spaCy text processing pipeline, optional): a custom spaCy text
+            processing pipeline
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(self, nlp=None, **kwargs):
+        self.nlp = nlp
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("nlp")
+    def run(self, nlp=None):
+        """
+        Task run method. Returns tagger component of spaCy pipeline.
+
+        Args:
+            - nlp (spaCy text processing pipeline, optional): a custom spaCy text
+                processing pipeline, must be provided if not
+                specified in construction
+
+        Returns:
+            - Tagger: spaCy Tagger object
+
+        """
+        return nlp.tagger
```

Comment:
I think that tasks should raise a value error if `nlp` is `None` (never passed in during run or initialization)

Reply:
Added logic and tests in latest commit